Support image descriptions using local AI model (#18475)
nvaccess/nvda: user_docs/en/userGuide.md

This option is only available if there are trusted fingerprints stored in your configuration.

#### Local Image Description Settings Panel {#LocalCaptionerSettings}

> This panel provides options to customize the behavior and default settings for the local image description feature. You can configure how and when the model used for describing images is loaded.

Reviewer: Please split sentences into new lines

```suggestion
This panel provides options to customize the behavior and default settings for the local image description feature.
You can configure how and when the model used for describing images is loaded.
```

This option is only available if there are trusted fingerprints stored in your configuration.

#### Local Image Description Settings Panel {#LocalCaptionerSettings}

This panel provides options to customize the behavior and default settings for the local image description feature.
You can configure how and when the model used for describing images is loaded.

##### Model Path {#LocalCaptionerSettingsModelPath}

Use this edit box to specify the path from which the model should be loaded.

> This allows you to use a custom model located on your local system.

Reviewer: I don't think we want a user visible setting like this.
For now, developers can change their settings config or write an add-on.
There should be no need for end users to change the path of their model directly.
Let's just ship with one model, and a proper user friendly model manager later

This option is only available if there are trusted fingerprints stored in your configuration.

> #### Local Image Description Settings Panel {#LocalCaptionerSettings}

Reviewer: please make sure this matches the settings panel

```suggestion
#### AI Image Descriptions Settings {#LocalCaptionerSettings}
```

> This panel provides options to customize the behavior and default settings for the local image description feature.

Reviewer: ```suggestion
This panel provides options to customize the behavior and default settings for the ["Image Captioner"](#LocalCaptioner).
```

> You can configure how and when the model used for describing images is loaded.

Reviewer: ```suggestion
```

> ##### enable image captioning {#LocalCaptionerSettingsLoadWhenInit}

Reviewer: ```suggestion
##### Enable image captioner {#LocalCaptionToggle}
```

| Send clipboard | None | Sends the contents of the clipboard to the remote computer. |
<!-- KC:endInclude -->

## Local Image Description {#LocalCaptioner}

NVDA supports generating image descriptions using a local model. This feature allows NVDA to describe image objects encountered during navigation. It supports shortcut key activation, and the result is automatically copied to the clipboard.

Note:
To reduce the NVDA installer size, the image captioning model is not bundled with NVDA by default.
You must first download the model via the NVDA menu → Tools → Model Manager before using this feature.

### Getting Started {#LocalCaptionerGettingStarted}

Before using this feature:

1. Open the Model Manager from the NVDA menu → Tools → Model Manager.
2. In the dialog, locate the Download button using the Tab key and press it.
3. Wait for the model to download. You can monitor progress in the Download Log area.
4. Once the model is ready, press the default shortcut `NVDA+Windows+,` to recognize the image currently navigated by NVDA.

> ### Model Download {#LocalCaptionerModelDownload}

Reviewer: hasn't the model manager been removed from this PR?

Note:
To reduce the NVDA installer size, the image captioning model is not bundled with NVDA by default.

### Getting Started {#LocalCaptionerGettingStarted}

Once the model is ready, press the default shortcut `NVDA+Windows+,` to recognize the image currently navigated by NVDA.

### AI Image Descriptions Key Commands Summary {#LocalCaptionerGestures}

<!-- KC:beginInclude -->

| Name                                  | Key                    | Description                                                                                |
| ------------------------------------- | ---------------------- | ------------------------------------------------------------------------------------------ |
| get an AI generated image description | `NVDA+Windows+,`       | Use a local model to generate a description of the image object currently navigated by NVDA. |
| toggle image captioning | Unassigned | load/unload the model from memory.                                       |

> | Shows the AI image descriptions settings  | Unassigned           | Opens the Local Image Description settings panel.                                          |

Reviewer: can you please remove all the extra white space

| Send `control+alt+delete` | None | Sends `control+alt+delete` to the controlled computer. |
<!-- KC:endInclude -->

> ## Local Image Description {#LocalCaptioner}

Reviewer: ```suggestion
## Image Captioner {#LocalCaptioner}
```

NVDA supports generating image descriptions using a local model. This feature allows NVDA to describe image objects encountered during navigation. It supports shortcut key activation, and the result is automatically copied to the clipboard.

> Note:
> To reduce the NVDA installer size, the image captioning model is not bundled with NVDA by default.
> 

Reviewer: ```suggestion
Note: An internet connection is required to enable and install the Image Captioner for the first time.
It is not included with the NVDA installer to reduce the installer size.
```

### Getting Started {#LocalCaptionerGettingStarted}

> Once the model is ready, press the default shortcut `NVDA+Windows+,` to recognize the image currently navigated by NVDA.

Reviewer: ```suggestion
Enable the "Image Captioner" in the ["AI Image Descriptions" settings panel](#).
Once the Image Captioner is ready, press the default shortcut `NVDA+Windows+,` to recognize the image currently focused.
```

### AI Image Descriptions Key Commands Summary {#LocalCaptionerGestures}

<!-- KC:beginInclude -->

| Name | Key | Description |
| ------------------------------------- | ---------------------- | ------------------------------------------------------------------------------------------ |
| get an AI generated image description | `NVDA+Windows+,` | Use a local model to generate a description of the image object currently navigated by NVDA. |
| toggle image captioning | Unassigned | load/unload the model from memory. |

> | Shows the AI image descriptions settings | Unassigned | Opens the Local Image Description settings panel. |

Reviewer: please make sure the names here match the final ones in the gestures. Please remove the references to models
